soroban-env-host/src/fees.rs

```diff
@@ -238,18 +241,20 @@ pub fn compute_write_fee_per_1kb(
         .write_fee_1kb_bucket_list_high
         .saturating_sub(fee_config.write_fee_1kb_bucket_list_low)
         .clamp_fee();
-    let bucket_list_size_before_reaching_target =
-        bucket_list_size_bytes.min(fee_config.bucket_list_target_size_bytes);
-    // Convert multipliers to i128 to make sure we can handle large bucket list
-    // sizes.
-    let mut write_fee_per_1kb = num_integer::div_ceil(
-        (fee_rate_multiplier as i128)
-            .saturating_mul(bucket_list_size_before_reaching_target as i128),
-        fee_config.bucket_list_target_size_bytes as i128,
-    )
-    .clamp_fee();
-    write_fee_per_1kb = write_fee_per_1kb.saturating_add(fee_config.write_fee_1kb_bucket_list_low);
-    if bucket_list_size_bytes > fee_config.bucket_list_target_size_bytes {
+    let mut write_fee_per_1kb: i64;
+    if bucket_list_size_bytes < fee_config.bucket_list_target_size_bytes {
+        // Convert multipliers to i128 to make sure we can handle large bucket list
+        // sizes.
+        write_fee_per_1kb = num_integer::div_ceil(
+            (fee_rate_multiplier as i128).saturating_mul(bucket_list_size_bytes as i128),
+            fee_config.bucket_list_target_size_bytes as i128,
+        )
+        .clamp_fee();
+        // no clamp_fee here
+        write_fee_per_1kb =
+            write_fee_per_1kb.saturating_add(fee_config.write_fee_1kb_bucket_list_low);
+    } else {
+        write_fee_per_1kb = fee_config.write_fee_1kb_bucket_list_high;
         let bucket_list_size_after_reaching_target =
             bucket_list_size_bytes - fee_config.bucket_list_target_size_bytes;
```

Comment:
this subtraction can underflow

Reply:
This subtraction can't underflow because it's in the else condition of `if bucket_list_size_bytes < fee_config.bucket_list_target_size_bytes`. I can still add the saturating_sub call to future proof if you think that makes sense.

Reply:
yeah, for future proofing I'd appreciate it. thanks!

Reply:
Done!